### PR TITLE
Fix #77 codebase remains Java 8 compatible

### LINKED ISSUE #77: Codebase must remain Java 8 compatible
`0.3.0` was published with Java 11 bytecode

```
$ jarviz bytecode show --gav dev.cdevents:cdevents-sdk-java:0.3.0
subject: cdevents-sdk-java-0.3.0.jar
Unversioned classes. Bytecode version: 53 (Java 9) total: 1
Unversioned classes. Bytecode version: 55 (Java 11) total: 339
```

consumers such as Spinaker remain in Java 8 compatibility mode.

### PR DESCRIPTION
**Pull request by aalmiray**
Fixes #77

### CODE REVIEW

**Comment by aalmiray** [comment]
```
$ jarviz bytecode show --file sdk/target/cdevents-sdk-java-0.3.1-SNAPSHOT.jar 
subject: cdevents-sdk-java-0.3.1-SNAPSHOT.jar
Unversioned classes. Bytecode version: 52 (Java 8) total: 341
Versioned classes 9. Bytecode version: 53 (Java 9) total: 1
```

File: sdk/src/main/java/dev/cdevents/spec/schemas/SchemaLoader.java [hunk 1]
```diff
@@ -37,7 +38,9 @@ public static String loadSchema(String schema) {
             }
 
             // load from filesystem next
-            return Files.readString(Paths.get(CDEventConstants.SCHEMA_FOLDER + "/" + schema));
+            try (Stream<String> lines = Files.lines(Paths.get(CDEventConstants.SCHEMA_FOLDER + "/" + schema))) {
+                return lines.collect(Collectors.joining("\n"));
+            }
```
Comment (afrittoli) [inline sdk/src/main/java/dev/cdevents/spec/schemas/SchemaLoader.java:43]: I guess the exception will surface to the outer `catch` - what is the value of the extra `try` here?

File: pom.xml [hunk 2]
```diff
@@ -34,7 +35,7 @@
 		<repository.url>git@github.com:${project.github.repository}.git</repository.url>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-		<project.build.outputTimestamp>2024-01-11T15:26:09Z</project.build.outputTimestamp>
+		<project.build.outputTimestamp>${git.commit.author.time}</project.build.outputTimestamp>
```
Comment (afrittoli) [inline pom.xml:38]: Nice :)

File: generator/pom.xml [hunk 3]
```diff
@@ -87,10 +88,7 @@
                             <sourcePaths>
                                 <sourcePath>${parent.project.dir}/spec/schemas/artifactpackaged.json</sourcePath>
                             </sourcePaths>
-                            <outputDirectory>${sdk.project.dir}/src/main/java</outputDirectory>
                             <targetPackage>dev.cdevents.models.artifact.packaged</targetPackage>
-                            <includeHashcodeAndEquals>true</includeHashcodeAndEquals>
-                            <includeToString>false</includeToString>
```
Comment (afrittoli) [inline generator/pom.xml:93]: How come these are going away, but the hash and equals method are still there? I guess it's the default behaviour to generate those?

File: sdk/pom.xml [hunk 4]
```diff
@@ -123,6 +123,7 @@
                         <configuration>
                             <overwriteExistingFiles>true</overwriteExistingFiles>
                             <failOnWarning>false</failOnWarning>
+                            <jvmVersion>9</jvmVersion>
```
Comment (afrittoli) [inline sdk/pom.xml:126]: Why is the target version 8 and the JVM version 9?

**Review by afrittoli — APPROVED** [review]
I don't really fully understand the changes here - I put some questions, but they are probably silly, feel free to ignore me 🙏 
For what I can understand it looks good to me.

**Comment by afrittoli** [comment]
@rjalander would you like to review this?

File: sdk/src/main/java/dev/cdevents/spec/schemas/SchemaLoader.java (hunk 1 above)
Comment (aalmiray) [reply sdk/src/main/java/dev/cdevents/spec/schemas/SchemaLoader.java:43]: The `try` block is used to auto close the stream.

File: pom.xml (hunk 2 above)
Comment (aalmiray) [reply pom.xml:38]: Sadly this property is overridden with every release and must be reverted to the expression after a release is posted.

File: generator/pom.xml (hunk 3 above)
Comment (aalmiray) [reply generator/pom.xml:93]: These settings are inherited from the common plugin configuration. Defining them on every execution was redundant.

**Comment by rjalander** [comment]
One quick question, Considering Spinnaker is with Java 11, (https://cdeliveryfdn.slack.com/archives/C030SKZ0F4K/p1707936366892439?thread_ts=1707859639.494349&cid=C030SKZ0F4K), 

Do we need this PR still to make Java 8 compatible?

**Comment by aalmiray** [comment]
Well, as much as I'd love for everyone to move to recent versions of Java, not everyone has the luxury to do so. With Java 8 having commercial support til 2030 there are still people holding up migration.

**Comment by rjalander** [comment]
Ok, so good to have a minimum supported version.

I think we need an update here https://github.com/cdevents/sdk-java/blob/main/DEVELOPMENT.md#install-tools for Java version.
If needed, can we have this change in the same PR?

**Comment by aalmiray** [comment]
There is no need for updating the Java version. Java 9 is still the minimum required because ModiTect needs it. 

The nice thing about the setup is that the produced bytecode is Java 8 compatible even though the build requires Java 9+

File: pom.xml (hunk 2 above)
Comment (afrittoli) [reply pom.xml:38]: I see - perhaps we could include that revert into the release pipeline?

**Review by rjalander — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (aalmiray) [reply pom.xml:38]: Right. It could be done with a regex. Let me have a look.

**Pull request merged**